sensors/sqs_sensor.py

```
@@ -98,50 +111,117 @@ def _may_setup_sqs(self):
+                                        assumed_role["Credentials"]["SecretAccessKey"],
+                                        assumed_role["Credentials"]["SessionToken"])
+
+        session = Session(
+            aws_access_key_id=self.credentials[account_id][0],
+            aws_secret_access_key=self.credentials[account_id][1],
+            aws_session_token=self.credentials[account_id][2]
+        )
+        self.sessions[account_id] = session
+
+    def _setup_sqs(self, session, account_id, region):
+        ''' Setup SQS resources'''
+        try:
+            if not self.sqs_res.get(account_id, None):
+                self.sqs_res[account_id] = {}
+
+            self.sqs_res[account_id][region] = session.resource('sqs', region_name=region)
 
-    def _get_queue_by_name(self, queueName):
-        ''' Fetch QUEUE by it's name create new one if queue doesn't exist '''
+            return self.sqs_res[account_id][region]
+        except UnknownEndpointError:
+            self._logger.error("Service 'sqs not available in region %s", region)
+            raise
+
+    def _get_info(self, queue):
+        ''' Retrieve the account ID and region from the queue URL '''
+        if queue.startswith('http://') or queue.startswith('https://'):
+            return queue.split('/')[3], queue.split('.')[1]
+        return self.account_id, self.aws_region
+
+    def _get_queue(self, queue, account_id, region):
+        ''' Fetch QUEUE by its name or URL and create new one if queue doesn't exist '''
         try:
-            return self.sqs_res.get_queue_by_name(QueueName=queueName)
+            sqs_res = self.sqs_res[account_id][region]
+        except KeyError:
+            try:
+                sqs_res = self._setup_sqs(self.sessions[account_id], account_id, region)
+            except KeyError:
+                self._logger.error('Session for account id %s does not exist', account_id)
+                return
+            except UnknownEndpointError:
```

It's not a good idea to ignore exceptions